pkg/controllermanager/controller/controllerregistration/controllerregistration_seed_control.go

```diff
@@ -478,12 +479,12 @@ func deployNeededInstallation(
 		kutil.SetMetaDataLabel(&controllerInstallation.ObjectMeta, common.RegistrationSpecHash, registrationSpecHash)
 
 		if controllerDeployment != nil {
-			deploymentSpecMap, err := convertObjToMap(controllerDeployment.Spec)
+			deploymentMap, err := convertObjToMap(controllerDeployment)
```

Comment:
Hmm, interesting.. this now in corporates also the metadata etc. of the `ControllerDeployment`, right? Not sure if that's a good idea, maybe we should still compute a checksum only based on `type` and `providerConfig`?

Reply:
That was actually one advantage of having a `spec` container, also when you look at the [mustIncreaseGeneration](https://github.com/gardener/gardener/pull/3995/files#diff-cdb4693983f8eb335e83f9782b23f9f76ab81ce8929e5fcc4fed6e08f7ff5ee8R59) function. I decided to take the entire `ControllerDeployment` object for the sake of simplicity and because we don't anticipate frequent metadata changes for this resource type. I almost vote for getting back the `spec`, so we don't have to explicitly list every field in such functions which is always bad for maintenance. WDYT?

Reply:
I think the impact is very little if you simply use `type` and `providerConfig` explicitly here. No need to have `spec` again just to be able to call to compute the hash sum more "conveniently". Currently, whenever the `.metadata.resourceVersion` changes (for whatever reason) the hash changes which is undesirable, I assume.

Reply:
It was just that one misses to adjust these places when new fields are added to the API (not because of implementation effort) which wouldn't be required if we had one spec.

Reply:
OK, sure, though, it's unlikely/not planned/didn't even happen since introduction of the `ControllerRegistration` (for the `.spec.deployment` field) that we add some new field that needs to be considered here.
You could still make sure that this place is not forgotten when the API is extended by adding a comment to the API definition of the `ControllerDeployment`.
Again, given that the frequency of code changes in this area are to be expected very minimal/quasi zero and that the impact of adapting these lines are quite minimal, I still suggest to only consider the relevant fields for the hash sum.